src/universal/decorators/socketWithPresence/reduxSocketOptions.js

```diff
@@ -0,0 +1,17 @@
+import {cashay, Transport} from 'cashay';
+import AuthEngine from 'universal/redux/AuthEngine';
+import socketCluster from 'socketcluster-client';
+
+const onConnect = (options, hocOptions, socket) => {
+  const sendToServer = request => {
+    return new Promise((resolve) => {
+      socket.emit('graphql', request, (error, response) => {
+        resolve(response);
+      });
+    });
+  };
+  const priorityTransport = new Transport(sendToServer);
```

Comment:
I still don't love this because it's a little brittle, but it works as long as `socketWithPresence` is the first place that the socket singleton is instantiated. Perhaps the right answer is to have socketCluster apply `AuthEngine` to a pre-existing singleton.

Basically, the `socket` singleton needs to read/write/remove our auth token. It does this via the custom `AuthEngine` I wrote. Since that `AuthEngine` uses redux, we need to supply it with our redux `store`. Since `redux-socket-cluster` has the store ready, i gave it a hook.